src/daft-functions-json/src/jq.rs

```diff
@@ -128,14 +128,20 @@ mod jq {
                             let res = compiled_filter
                                 .run((Ctx::new([], &inputs), json.into()))
                                 .map(|result| {
-                                    result.map(|v| v.to_string()).map_err(|e| {
+                                    result.map_err(|e| {
                                         DaftError::ComputeError(format!(
-                                            "Error running json query ({query}): {e}"
+                                            "Error running json filter ({filter}): {e}"
                                         ))
                                     })
                                 })
                                 .collect::<DaftResult<Vec<_>>>()
-                                .map(|values| Some(values.join("\n")));
+                                .map(|values| {
+                                    match values.len() {
+                                        0 => None,
+                                        1 => Some(values[0].to_string()),
+                                        _ => Some(Val::arr(values).to_string()), // need multiple matches to still be a valid JSON string
```

Comment:
This is an interesting one ... _technically_ jq says we return newline delimited values .. which totally makes for stdin/stdout. However we say the returned value is "valid json" which is why I wrapped it up. I could point out some issues with this, but ultimately I think using some collection here is a good idea.